File: source/Java/ScrapingManager/ScrapManager.Portal/src/main/java/Entity/Parameter.java
package Entity;

import java.lang.reflect.Field;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Date;
import java.util.List;

import javax.json.Json;
import javax.json.JsonObjectBuilder;
import Broker.BrokerUnit;
import entity.KeyNode;
import entity.RequestData;

public class Parameter extends RequestData implements Cloneable {

	public Parameter(int mallkey, String key) {
		super(mallkey, key);
		super.setCreatedDate(new Date());
	}

	private BrokerUnit broker;
	private Date ping;

	public BrokerUnit getBroker() {
		return broker;
	}

	public void setBroker(BrokerUnit broker) {
		this.broker = broker;
	}

	public Date getPing() {
		return ping;
	}

	public void setPing(Date ping) {
		this.ping = ping;
	}

	public String toJson() {
		JsonObjectBuilder builder = Json.createObjectBuilder();
		addJson(builder, "Key", getKey());
		addJson(builder, "ApiKey", getApikey());
		addJson(builder, "MallCD", getMallKey());
		addJson(builder, "Id1", getId1());
		addJson(builder, "Id2", getId2());
		addJson(builder, "Id3", getId3());
		addJson(builder, "Pw1", getPw1());
		addJson(builder, "Pw2", getPw2());
		addJson(builder, "Pw3", getPw3());
		addJson(builder, "Option1", getOption1());
		addJson(builder, "Option2", getOption2());
		addJson(builder, "Option3", getOption3());
		addJson(builder, "ScrapType", getScraptype());
		addJson(builder, "Sdate", getSdate());
		addJson(builder, "Edate", getEdate());
		addJson(builder, "Exec", getExec());
		return builder.build().toString();
	}

	private JsonObjectBuilder addJson(JsonObjectBuilder builder, String name, int value) {
		builder.add(name, value);
		return builder;
	}

	private JsonObjectBuilder addJson(JsonObjectBuilder builder, String name, String value) {
		if (value == null) {
			builder.addNull(name);
		} else {
			builder.add(name, value);
		}
		return builder;
	}

	public Parameter clone(int mallkey, String key) {
		try {
			Parameter clonedata = (Parameter) super.clone();
			setField(clonedata, "mallkey", mallkey);
			setField(clonedata, "key", key);
			return clonedata;
		} catch (CloneNotSupportedException e) {
			throw new RuntimeException(e);
		}
	}

	private void setField(Object obj, String name, Object value) {
		try {
			Field field = getAllFields(Parameter.class).stream().filter(x -> x.getName().equals(name)).findFirst().get();
			field.setAccessible(true);
			field.set(obj, value);
		} catch (IllegalAccessException e) {
			throw new RuntimeException(e);
		}
	}

	private List<Field> getAllFields(Class<?> type) {
		List<Field> fields = new ArrayList<>();
		getAllFields(fields, type);
		return fields;
	}

	private List<Field> getAllFields(List<Field> fields, Class<?> type) {
		if (type.getSuperclass() != null) {
			getAllFields(fields, type.getSuperclass());
		}
		fields.addAll(Arrays.asList(type.getDeclaredFields()));
		return fields;
	}

	public KeyNode getKeyNode() {
		KeyNode node = new KeyNode(getKey());
		node.setMallkey(getMallKey());
		return node;
	}
}

File: source/C#/Scraping/WebScraping/Parameter.cs
using System;
using Newtonsoft.Json;

namespace WebScraping
{
    public class Parameter
    {
        public String Key { get; set; }
        public int MallCD { get; set; }
        public String Id1 { get; set; }
        public String Id2 { get; set; }
        public String Id3 { get; set; }
        public String Pw1 { get; set; }
        public String Pw2 { get; set; }
        public String Pw3 { get; set; }
        public String Option1 { get; set; }
        public String Option2 { get; set; }
        public String Option3 { get; set; }
        public String Sdate { get; set; } // 매출내역, 정산내역, 반품율 스크랩 기간 yyyyMM 데이터
        public String Edate { get; set; } // 매출내역, 정산내역, 반품율 스크랩 기간 yyyyMM 데이터
        public ExecType Exec { get; set; } // 실행할 Scraper : 0 -> Only1Scraper, 1 -> Scraper
        public ScrapType ScrapType { get; set; } // 요청타입
        public DateTime Starttime { get; set; } // Scraper 실행시간
        public DateTime Pingtime { get; set; } // Scraper 중간 체크 시간
        public String State { get; set; } // Scraper 상태 표시 -> Only1Scraper는 ScrapType와 동일하게 움직여서 99번 요청이 들어 오는 경우가 아니면 필요 없을듯 합니다. Only1Scraper는 99번을 한번에 처리 할수 있도록 함. Exec가 Only1Scraper이면 manager가 각 Scraptype으로 나누지 않아도 됨(추후 적용 예정) : Only1Scraper는 월별 합계금액으로 데이터 생성하므로 가능함

        public override string ToString()
        {
            return JsonConvert.SerializeObject(this);
        }

        public string ToJson()
        {
            return ToString();
        }

        public static Parameter Build(string json)
        {
            return JsonConvert.DeserializeObject<Parameter>(json);
        }
    }
}
